packages/rspack/src/config/builtins.ts

```diff
@@ -373,7 +375,10 @@ export function resolveBuiltinsOptions(
 		pluginImport: resolvePluginImport(builtins.pluginImport),
 		relay: builtins.relay
 			? resolveRelay(builtins.relay, contextPath)
-			: undefined
+			: undefined,
+		codeGeneration: builtins.codeGeneration
```

Comment:
I think you could put the undefined logic inside `resolveCodeGeneration` function.